### PR TITLE
[RFC] -Wconversion for buffer.c

### PR DESCRIPTION
**Pull request by Adam-K-P**
I eliminated the wconflicts warnings in buffer.c.  This is my first commit and I'm new to git so please be nice :)

### CODE REVIEW

**Comment by KamranMackey** [comment]
A

**Comment by KamranMackey** [comment]
@Adam-K-P Looks good. :+1:

File: src/nvim/buffer.c [hunk 1]
```diff
@@ -2629,7 +2629,7 @@ void maketitle(void)
 
   if (p_title) {
     if (p_titlelen > 0) {
-      maxlen = p_titlelen * Columns / 100;
+      maxlen = (int)(p_titlelen * Columns) / 100;
```
Comment (bfredl) [inline src/nvim/buffer.c]: why not `(int)(p_titlelen * Columns / 100)` as it is the type of the end result that matters? (I know 32-bit division is faster, but this is hardly perfomance critical section)

**Comment by bfredl** [comment]
you need convert to `(int)base` on line 3619.

File: src/nvim/buffer.c [hunk 2]
```diff
@@ -3045,7 +3045,7 @@ int build_stl_str_hl(
         out_p = out_p - n + 1;
         /* Fill up space left over by half a double-wide char. */
         while (++group_len < item[groupitem[groupdepth]].minwid)
-          *out_p++ = fillchar;
+          *out_p++ = (char_u)fillchar;
```
Comment (bfredl) [inline src/nvim/buffer.c]: this ain't needed as you already changed to `char_u fillchar` above

File: src/nvim/buffer.c [hunk 3]
```diff
@@ -3066,7 +3066,7 @@ int build_stl_str_hl(
         if (min_group_width < 0) {
           min_group_width = 0 - min_group_width;
           while (group_len++ < min_group_width && out_p < out_end_p)
-            *out_p++ = fillchar;
+            *out_p++ = (char_u)fillchar;
```
Comment (bfredl) [inline src/nvim/buffer.c]: same

File: src/nvim/buffer.c [hunk 4]
```diff
@@ -3086,7 +3086,7 @@ int build_stl_str_hl(
 
           // Prepend the fill characters
           for (; group_len > 0; group_len--) {
-            *t++ = fillchar;
+            *t++ = (char_u)fillchar;
```
Comment (bfredl) [inline src/nvim/buffer.c]: same

**Comment by bfredl** [comment]
LGTM after comments and rebase.

**Pull request closed without merging**